src/coreclr/src/jit/codegenxarch.cpp

```diff
@@ -8492,11 +8493,12 @@ void CodeGen::genProfilingEnterCallback(regNumber initReg, bool* pInitRegZeroed)
     // Push the profilerHandle
     if (compiler->compProfilerMethHndIndirected)
     {
-        GetEmitter()->emitIns_AR_R(INS_push, EA_PTR_DSP_RELOC, REG_NA, REG_NA, (ssize_t)compiler->compProfilerMethHnd);
+        GetEmitter()->emitIns_AR_R(INS_push, EA_PTR_DSP_RELOC, REG_NA, REG_NA,
+                                   (target_ssize_t)(size_t)compiler->compProfilerMethHnd);
```

Comment:
Same comment: I don't think this is correct. For x64 hosted targeting x86, I presume compProfilerMethHnd will be a 64-bit pointer than needs to be relocated, but this will truncated it to 32 bits.